### ISSUE TITLE
Wishlist highlighting incorrect perks

### ISSUE THREAD

**Issue by buncieboy343** [DestinyItemManager/DIM#6412]
Version 6.46.0 (release)

Chrome

Using voltron for wishlisht, DIM is not showing the correct perks for this weapon (the perk combination recommended is not possible). From searching voltron it appears that it doesn't recommend Full Auto Trigger System on this weapon, but does recommend slideshot/outlaw. 

![image](https://user-images.githubusercontent.com/76978475/103625952-80eeeb80-4f33-11eb-944b-390050e9854d.png)

**Comment by buncieboy343** [comment]
Still sorting my vault and saw this one, again it is on the wishlist with mulligan, but this time no perks are highlighted in the third column.

![image](https://user-images.githubusercontent.com/76978475/103628666-3d967c00-4f37-11eb-8ca5-092b356d23ef.png)

**Comment by kyleshay** [comment]
Aah sigh. Thanks for the report. This may likely be due do the selective component loading on the inventory page. I think I fixed it for some cases, but it looks like the issue is persisting. I may just rollback the change and dig deeper to see what's going on.

**Comment by buncieboy343** [comment]
Sorry to be the bearer of bad news! Its the only 2 I noticed out of a full vault if that helps? Are further examples of any use or do you have all the info you need?

**Comment by bhollis** [comment]
Is this still happening?

**Comment by buncieboy343** [comment]
@bhollis Both guns mentioned have since been yeeted from my vaulted, so cannot confirm. Haven't noticed any other examples so far, but could be some buried in there. Sorry that it's not much help!

**Comment by buncieboy343** [comment]
![image](https://user-images.githubusercontent.com/76978475/108265559-d6d4c700-7160-11eb-8335-c71761f2d7c2.png)

Just went through and checked the 150 ish guns in my vault that are part of the wishlist, this was the only example I could find where it didn't highlight 1 perk in each column. Not sure if related, or is even an issue.

**Comment by delphiactual** [comment]
wishlist do not have to specify every column, there is likely a wishlist for smg which do not care about scopes in your selected file.

in fact here is the snippet from voltron
```
dimwishlist:item=1786797708&perks=106909392,3643424744,3708227201#notes:Chevy's Super Pog PvE Loot List. Recommended MW -  Handling/Reload
dimwishlist:item=1786797708&perks=1087426260,3643424744,3708227201#notes:Chevy's Super Pog PvE Loot List. Recommended MW -  Handling/Reload
dimwishlist:item=1786797708&perks=106909392,4071163871,3708227201#notes:Chevy's Super Pog PvE Loot List. Recommended MW -  Handling/Reload
dimwishlist:item=1786797708&perks=1087426260,4071163871,3708227201#notes:Chevy's Super Pog PvE Loot List. Recommended MW -  Handling/Reload
```

**Comment by buncieboy343** [comment]
Ok, so confirmed not an issue, but don't have the original guns to verify if that issue is resolved.

**Comment by buncieboy343** [comment]
![image](https://user-images.githubusercontent.com/76978475/108448189-2dbdc780-7259-11eb-8347-adeeb583e641.png)

Just did some testing and created a fake roll of the toil and trouble in the API response, and got the same original issue again. Could this be re-opened?

**Comment by delphiactual** [comment]
please post your fake roll

**Comment by buncieboy343** [comment]
I faked the API request, which parts do you need as the whole request is rather large, sorry new to this API/project

**Comment by delphiactual** [comment]
at least the perkhashes

**Comment by delphiactual** [comment]
from `voltron.txt`
```
dimwishlist:item=2957367743&perks=1047830412,791862061,3161816588,2117683199
dimwishlist:item=2957367743&perks=1332244541,791862061,3161816588,2117683199
dimwishlist:item=2957367743&perks=1047830412,1087426260,3161816588,2117683199
dimwishlist:item=2957367743&perks=1332244541,1087426260,3161816588,2117683199
dimwishlist:item=2957367743&perks=1047830412,106909392,3161816588,2117683199
dimwishlist:item=2957367743&perks=1332244541,106909392,3161816588,2117683199
dimwishlist:item=2957367743&perks=1047830412,791862061,3300816228,2117683199
dimwishlist:item=2957367743&perks=1332244541,791862061,3300816228,2117683199
dimwishlist:item=2957367743&perks=1047830412,1087426260,3300816228,2117683199
dimwishlist:item=2957367743&perks=1332244541,1087426260,3300816228,2117683199
dimwishlist:item=2957367743&perks=1047830412,106909392,3300816228,2117683199
dimwishlist:item=2957367743&perks=1332244541,106909392,3300816228,2117683199
```

**Comment by delphiactual** [comment]
matching this one specifically
```
dimwishlist:item=2957367743&perks=1047830412,1087426260,3161816588,2117683199
```

**Comment by buncieboy343** [comment]
Yeah the issue wasn't that it didn't match, its that it highlights the incorrect (technically impossible) perks.

"From searching voltron it appears that it doesn't recommend Full Auto Trigger System on this weapon, but does recommend slideshot/outlaw."

**Comment by delphiactual** [comment]
https://data.destinysets.com/i/InventoryItem:2117683199 is full auto trigger system

**Comment by delphiactual** [comment]
![image](https://user-images.githubusercontent.com/4798491/108450554-ac4b4280-7222-11eb-84dd-375375b981fd.png)

If you think `voltron.txt` is wrong then the correct place for your issue would be at https://github.com/48klocs/dim-wish-list-sources

**Comment by delphiactual** [comment]
Recommended by panda_paxxy for forsaken here: https://docs.google.com/spreadsheets/d/1YWX2NCw8joV0T15ivF7Hu98iMC1Za4lzMWpwJiMx6q8/edit#gid=1967975290

**Comment by buncieboy343** [comment]
Again not an issue with voltron, issue belongs here. If its matching with "dimwishlist:item=2957367743&perks=1047830412,1087426260,3161816588,2117683199" then that is wrong as you can't equip that combination on this roll. It appears to be matching both of these
"dimwishlist:item=2957367743&perks=1047830412,1087426260,3161816588,1168162263
dimwishlist:item=2957367743&perks=1332244541,1087426260,3161816588,1168162263" but instead highlighting the first one, which isn't possible to equip on this roll.

**Comment by delphiactual** [comment]
@48klocs

**Comment by buncieboy343** [comment]
Does the issue make sense now? Sorry its 3AM here so I'm probably not explaining myself very well.

**Comment by delphiactual** [comment]
No. It is matching what I am telling you. Thumbs up do not indicate that you must equip for god roll, as some god rolls were allowable with 2 perks interchangeable in the god roll as it reduces the number of weapons required from 2 to 1. I will defer this to 48klocs though as this is their domain.

**Comment by buncieboy343** [comment]
Unsure if the second weapon (braytech werewolf roll with only 3 highlighted) is also covered by the same issue. I will take a stab at faking that roll tomorrow, as on Voltron all rolls for that weapon require 4 perks, where is in the image above only 3 get highlighted.

![image](https://user-images.githubusercontent.com/76978475/108452998-76797e80-7261-11eb-9b4f-1e63efa8669c.png)
Looks like you are correct that is is matching on that original one, changed the 4th column to snapshot sights and it still matched.
Correct me if I am wrong but that means its one of two cases: 
- the roll is intentionally there as it reduces the amount of guns from 2 to 1, at which point it might be worth adding a note either on voltron or within DIM explaining why its matched, as in the thousands of matching rolls I've recieved whilst playing these two are the only ones to have this occur.
- the roll is a mistake, at which point its a simple voltron issue

Only thing that is strange to me is if we remove the full auto system it loses its god roll status (as seen below).  It makes sense as you mentioned it reduces the need for 2 guns to 1, but surely either one of them would still be a god roll in their own right? Again this is a voltron issue, so probably out of scope for here.

![image](https://user-images.githubusercontent.com/76978475/108453408-5d250200-7262-11eb-9afb-24756bd141bf.png)

Either way seems like you are right that correct action is to defer to @48klocs.

**Comment by buncieboy343** [comment]
Seems like issue with the braytech werewolf persists, image below.

Looks like a match of this line dimwishlist:item=528834068&perks=1482024992,1561002382,3513791699,4082225868, but mulligan isn't highlighted.

![image](https://user-images.githubusercontent.com/76978475/108454770-ea695600-7264-11eb-9fa4-9c91fc82012b.png)

**Comment by 48klocs** [comment]
It feels like there's a number of maybe-issues wrapped up in here.

If you think you've found an issue with how DIM is displaying/not displaying perks for rolls, I'd invite you to...
- create a simple reproduction wish list file (just one line!) and load that alone
- use https://data.destinysets.com/ to verify that all the perks are what you think they are
- let us know what's acting up
- and that it's not acting up during/after a maintenance period

If you think you've found an issue with the wish list file itself, feel free to log an issue in https://github.com/48klocs/dim-wish-list-sources - I've made mistakes before, and if you're using panda as a benchmark, they update their spreadsheet from time to time (and I rarely reflect those changes).

**Comment by buncieboy343** [comment]
Tested the braytech werewolf issue using a wishlist consisting of one line (dimwishlist:item=528834068&perks=1482024992,1561002382,3513791699,4082225868) and DIM still isn't highlighting mulligan. Not sure if issue is specifically with mulligan, as I have no other rolls from the wishlist with mulligan. As mentioned before I had to fake the response to bungies API call to get this, as I have deleted the item, but had the roll back in January.

![image](https://user-images.githubusercontent.com/76978475/108599791-331b2f00-738b-11eb-953e-a1f545bea6a3.png)

Only issue I can see on DIMs side with toil and trouble is that if it is a mistake (full auto being in the 4th column) on voltrons side DIM doesn't check which column it matches to on the weapon. I assume this would be too much hassle to fix as I understand the wishlist supports leaving columns blank. Either way I'll raise a separate issue on Voltron to confirm if slideways+full auto is intended.
